crates/ark/src/modules/positron/viewer.R

```diff
@@ -6,11 +6,28 @@
 #
 
 options("viewer" = function(url, height = NULL, ...) {
+    # Normalize paths for comparison. This is necessary because on e.g. macOS,
+    # the `tempdir()` may contain `//` or other non-standard path separators.
+    normalizedPath <- normalizePath(url, mustWork = FALSE)
+    normalizedTempdir <- normalizePath(tempdir(), mustWork = FALSE)
+
     # Is the URL a temporary file?
-    if (startsWith(url, tempdir())) {
+    if (startsWith(normalizedPath, normalizedTempdir)) {
+        # Use the filename as the label, unless it's an index file, in which
+        # case use the directory name.
+        fname <- tolower(basename(url))
+        if (identical(fname, "index.html") || identical(fname, "index.htm")) {
+            fname <- basename(dirname(url))
+        }
+
+        # R HTML widgets get printed to temporary files starting with the name
+        # "viewhtml". This makes an ugly label, so we give it a nicer one.
+        if (startsWith(fname, "viewhtml")) {
+            fname <- "R HTML widget"
+        }
+
         # If so, open it in the HTML viewer.
-        .ps.Call("ps_html_viewer", url)
-        # TODO: handle `height` for HTML viewer
+        .ps.Call("ps_html_viewer", url, fname, height, FALSE)
```

Comment:
Should probably do some R side validation of `height` here?

Like, in other places i saw this

```
if (is.null(height)) {
        height <- 0
    }
```

Reply:
In fact, isn't a `NULL` height a problem here right now? Can that even be converted to an i32? If so I'd be surprised

Reply:
Oh I see

```r
                    height: match height {
                        Ok(height) => height.into(),
                        Err(_) => 0,
                    },
```

That is probably catching the _error_ resulting from a `NULL` height and it just happens to get mapped to `0`. That definitely seems like a conversion to handle on the R side then

Reply:
I'll add handling on the R side too! But I'm going the leave the Rust bit in as a layer of defense (anyone can call this function if they try hard enough and I don't think a type error should cause a panic here)